src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -26,17 +55,170 @@ const ActorInit Shot_Sun_InitVars = {
+        this->actor.flags |= 0x8000000;
+    } else {
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+        this->actionFunc = &ShotSun_HyliaSunUpdate;
+        this->actor.flags &= ~1;
+    }
+}
+
+void ShotSun_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ShotSun* this = THIS;
+    s32 temp_v0;
+
+    temp_v0 = this->actor.params & 0xFF;
+    if (temp_v0 != 0x40 && temp_v0 != 0x41) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+void ShotSun_FairyCountdown(ShotSun* this, GlobalContext* globalCtx) {
+    s32 temp_v0 = this->actor.params & 0xFF;
+    s32 sp38;
+
+    if (this->spawnTimer > 0) {
+        this->spawnTimer -= 1;
+        return;
+    }
+
+    switch (temp_v0) {
+        case 0x40:
+            sp38 = 7;
+            break;
+        case 0x41:
+            sp38 = 7;
+            break;
+    }
+
+    Actor_Spawn(
+        &globalCtx->actorCtx, globalCtx, ACTOR_EN_ELF,
+        this->actor.initPosRot.pos.x, this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z,
+        0, 0, 0,
+        sp38
+    );
+
+    Actor_Kill(&this->actor);
+}
+
+void ShotSun_StartFairyCountdown(ShotSun* this, GlobalContext* globalCtx) {
+    if ((func_8005B198() == this->actor.type) || (this->spawnTimer != 0)) {
+        this->actionFunc = &ShotSun_FairyCountdown;
+        this->spawnTimer = 50;
+
+        Actor_Spawn(
+            &globalCtx->actorCtx, globalCtx, ACTOR_DEMO_KANKYO,
+            this->actor.initPosRot.pos.x, this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z,
+            0, 0, 0,
+            0x11
+        );
+
+        func_80078914(&this->actor.projectedPos, NA_SE_EV_TRE_BOX_APPEAR);
+    }
+}
+
+void func_80BADF0C(ShotSun* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    s32 temp_t0 = this->actor.params & 0xFF;
+
+    if (22500.0f < func_800CB650(&this->actor.posRot.pos, &player->actor.posRot.pos)) {
+        this->unk_1A4[0] = 0;
+    } else {
+        if (this->unk_1A4[0] == 0) {
+            if (((s32)player->stateFlags2 << 7) >= 0) {
+                player->stateFlags2 |= 0x800000;
+                return;
+            } else {
+                this->unk_1A4[0] = 1;
+            }
+        }
+        if (this->unk_1A4[0] == 1) {
+            func_8010BD58(globalCtx, 1U);
+            this->unk_1A4[0] = 2;
+            return;
+        }
+        
+        if (this->unk_1A4[0] == 2 && globalCtx->msgCtx.unk_E3EE == 4) {
+            if (((temp_t0 == 0x40) && (*(u16*)(&globalCtx->msgCtx.unk_E3E8[4]) == 9))
+                || ((temp_t0 == 0x41) && (*(u16*)(&globalCtx->msgCtx.unk_E3E8[4]) == 0xB))) {
+                    this->actionFunc = &ShotSun_StartFairyCountdown;
+                    func_80080480(globalCtx, this);
+                    this->spawnTimer = 0;
+            } else {
+                this->unk_1A4[0] = 0;
+            }
+
```

Comment:
super tiny but id remove this white space. clang might even do this for you which you should run if you havent already.
youll need to install clang-tidy and clang-format and run ./format.sh from root